lineapy/instrumentation/control_flow_tracker.py

```diff
@@ -0,0 +1,23 @@
+from dataclasses import dataclass, field
+from typing import List, Optional
+
+from lineapy.data.types import LineaID
+
+
+@dataclass
+class ControlFlowTracker:
+
+    control_flow_node_stack: List[LineaID] = field(default_factory=list)
+
+    def push_node(self, control_node_id: LineaID) -> None:
+        self.control_flow_node_stack.append(control_node_id)
+
+    def pop_node(self) -> LineaID:
+        return self.control_flow_node_stack.pop()
+
+    def current_control_dependency(self) -> Optional[LineaID]:
+        return (
```

Comment:
switch to context manager in the next PR maybe. no need to have that creep in this one.

Reply:
already added